Android. remember last channel value saved on tt files

amirmahdifard opened the pull request:
from now on, when exporting a tt file, doesn't matter for a single server or the server list, the remember last channel checkbox's value will also be saved on tt files as how as the user have set them and will be loaded when importing tt file, so users do not have to turn it on or off again for any server, their settings are restored as how as it was before.
this fixes https://github.com/BearWare/TeamTalk5/issues/2554

amirmahdifard commented:
@bear101 Do you want me to also save nicknames? currently nicknames for the servers will be lost in tt files. LOL I surely know that saving it is a good idea and no asking is needed, but I was a bit layzy, so I wanted to ask you to get some motivation, then i'll add. LOL

bear101 commented on Client/TeamTalkAndroid/src/main/java/dk/bearware/gui/Utils.java:
```diff
@@ -369,6 +368,9 @@ public static Vector<ServerEntry> getXmlServerEntries(String xml) {
                     Node joinnode = joinlist.item(0);
                     if (joinnode.getNodeType() == Node.ELEMENT_NODE) {
                         Element joinelement = (Element) joinnode;
+                        NodeList rememberLastChannelnode = hostelement.getElementsByTagName("rememberLastChannel");
```
can you rename it to "join-last-channel"? Then it matches the name from desktop client. Also other xml-tags are lower-case with dash.

bear101 commented:
Yes, also put nickname please

amirmahdifard commented:
in desktop clients this value is not even saved in tt files as I looked at it, But ok, sure.

amirmahdifard commented:
@bear101 ok. I've added nickname as wel, so now tt support is complete in android devices. All user preferences will bring in tt files. And also, I've changed the tag name to "joinlastchannel" As you asked me too, But I didn't add the dashes, because even tho I was not sure that it will work orr no, I tryed it but it gets compilation errors, and also, I've looked at all other tags, and didn't see even a single dash in any of them. So, this is now as perfect as possible. Thanks. When this is merged, i'll start to work on other things again.

amirmahdifard commented:
@bear101 please look at serverlist activity.java file, line 167, In this part
    @Override
    protected void onStart() {
        super.onStart();

        if ((serverentry != null) && serverentry.rememberLastChannel) {
            saveServers();
        }
This doesn't makes sence for me why we check && entry.rememberLastChannel in this section. Do you know why? or if it is incorrectly added, I can remove it and just call if serverentry !=nul save servers, which Is correct. Entry.RememberLastChannel hear is completely unrelated. Thanks

amirmahdifard commented:
also exists on server entry activity.java, line 126
    @Override
    protected void onStart() {
        super.onStart();
        
        if ((serverentry != null) && serverentry.rememberLastChannel) {
            showServer(serverentry);
            serverentry = null;
        }
tell me is it ok to remove these entry.rememberLastChannel from these checks. Thanks

bear101 commented:
> @bear101 please look at serverlist activity.java file, line 167, In this part @OverRide protected void onStart() { super.onStart();
> 
> ```
>     if ((serverentry != null) && serverentry.rememberLastChannel) {
>         saveServers();
>     }
> ```
> 
> This doesn't makes sence for me why we check && entry.rememberLastChannel in this section. Do you know why? or if it is incorrectly added, I can remove it and just call if serverentry !=nul save servers, which Is correct. Entry.RememberLastChannel hear is completely unrelated. Thanks

I also don't know. I don't see why you would save the server list in onStart(). I would assume the server list is unchanged.

bear101 commented:
> also exists on server entry activity.java, line 126 @OverRide protected void onStart() { super.onStart();
> 
> ```
>     if ((serverentry != null) && serverentry.rememberLastChannel) {
>         showServer(serverentry);
>         serverentry = null;
>     }
> ```
> 
> tell me is it ok to remove these entry.rememberLastChannel from these checks. Thanks

Hm, it would seem `rememberLastChannel` is always true.

amirmahdifard commented:
@bear101 So can I remove these checks please?

amirmahdifard commented:
@bear101 I won't touch if you don't want me to, but, I suggest to remove that to keep the code more cleaner. Anyway, tell me should I touch them, or no and just merge the pr? thanks for answering.

bear101 commented:
> @bear101 I won't touch if you don't want me to, but, I suggest to remove that to keep the code more cleaner. Anyway, tell me should I touch them, or no and just merge the pr? thanks for answering.

I don't see the purpose of checking `rememberLastChannel` on these `if` statements. But please do thorough testing if you remove them.

amirmahdifard commented:
ok, I'll test them properly before I remove, modify or add anything, and everytime I was doing that. You won't find any problems from the features i've added or modifyed

bear101 commented on Client/TeamTalkAndroid/src/main/java/dk/bearware/gui/Utils.java:
```diff
@@ -369,6 +371,9 @@ public static Vector<ServerEntry> getXmlServerEntries(String xml) {
                     Node joinnode = joinlist.item(0);
                     if (joinnode.getNodeType() == Node.ELEMENT_NODE) {
                         Element joinelement = (Element) joinnode;
+                        NodeList joinlastchannelnode = hostelement.getElementsByTagName("joinlastchannel");
```
Please rename to "join-last-channel"

amirmahdifard commented:
@bear101 are you really sure? because any other tags in android that I see doesn't have dashes. So are you sure this one should have?

amirmahdifard commented:
@bear101 I removed unused code that I told you before, And i tested everything with it, and made sure that code was never even used, and doesn't cause any bug

amirmahdifard commented:
@bear101 I don't understand what you say. Do you want me to bring the if statements back and just remove the entry.remember last channel from it? I'm not sure that's what you want from your comment, But ok i'll bring them back

amirmahdifard commented:
@bear101 done. Actually this is better, because we make sure any new bug or crash won't appear

amirmahdifard commented:
@bear101 As I said, I didn't see any dashes in tags in android. So, today, please tell me your final choice, will you rename that to joinlastchannel, or I should rename to join-last-channel. And then, this pr is ready to merge.

bear101 commented:
Please rename to join-last-channel

amirmahdifard commented:
@bear101 done
